src/posix/platform/misc.cpp

```diff
@@ -40,14 +40,14 @@
 #include "common/code_utils.hpp"
 #include "common/logging.hpp"
 
-static otPlatResetReason   sPlatResetReason   = OT_PLAT_RESET_REASON_POWER_ON;
+otPlatResetReason          gPlatResetReason   = OT_PLAT_RESET_REASON_POWER_ON;
```

Comment:
I tried to not use a global variable but it seems not that simple to do so. 

I think `otPlatGetResetReason` are supposed to be defined by platform code and used by core code. So, it not proper to define a new method (i.e. otPlatSetResetReason) only to be called by core code. 

We are already using other global variables to be accessed by multiple platform sources. That's why I think using global variable is acceptable here.

Reply:
> I tried to not use a global variable but it seems not that simple to do so.
> 
> I think `otPlatGetResetReason` are supposed to be defined by platform code and used by core code. So, it not proper to define a new method (i.e. otPlatSetResetReason) only to be called by platform code.
> 
> We are already using other global variables to be accessed by multiple platform sources. That's why I think using global variable is acceptable here.

I see and since the precedent has already been set I'm good with the change.